Core/Src/main.c

```diff
@@ -65,41 +67,86 @@ void SystemClock_Config(void);
 int main(void)
 {
   /* USER CODE BEGIN 1 */
-
-  /* USER CODE END 1 */
-
-  /* MCU Configuration--------------------------------------------------------*/
-
-  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
-  HAL_Init();
-
-  /* USER CODE BEGIN Init */
-
-  /* USER CODE END Init */
-
-  /* Configure the system clock */
-  SystemClock_Config();
-
-  /* USER CODE BEGIN SysInit */
-
-  /* USER CODE END SysInit */
-
-  /* Initialize all configured peripherals */
-  MX_GPIO_Init();
-  MX_USART2_UART_Init();
-  /* USER CODE BEGIN 2 */
-
-  /* USER CODE END 2 */
-
-  /* Infinite loop */
-  /* USER CODE BEGIN WHILE */
-  while (1)
-  {
-    /* USER CODE END WHILE */
-
-    /* USER CODE BEGIN 3 */
-  }
-  /* USER CODE END 3 */
+    //Transmit Bytes
+    //First byte: 00000001 -> 1
+    //Second byte: 1000000 -> 128
+    //Third byte: doesn't matter -> 0
+    uint8_t tx_data[3] = {1, 128, 0};
+    //Receive Bytes
+    uint8_t rx_data[3] = {0};
+    const uint8_t mask = 3;
+    uint16_t secondByte;
+    uint16_t thirdByte;
+    uint16_t counts;
+    /* USER CODE END 1 */
+
+    /* MCU Configuration--------------------------------------------------------*/
+
+    /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+    HAL_Init();
+
+    /* USER CODE BEGIN Init */
+
+    /* USER CODE END Init */
+
+    /* Configure the system clock */
+    SystemClock_Config();
+
+    /* USER CODE BEGIN SysInit */
+
+    /* USER CODE END SysInit */
+
+    /* Initialize all configured peripherals */
+    MX_GPIO_Init();
+    MX_USART2_UART_Init();
+    MX_SPI1_Init();
+    MX_TIM1_Init();
+    /* USER CODE BEGIN 2 */
+    //Status
+    HAL_StatusTypeDef hal_status;
+    //Pull CS High
+    HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+    /* USER CODE END 2 */
+
+    /* Infinite loop */
+    /* USER CODE BEGIN WHILE */
+    while (1)
+    {
+        /* USER CODE END WHILE */
+	    //Pull CS Low
+	    HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	    //Initiate Communication with MCP 3004
+	    hal_status = HAL_SPI_TransmitReceive(&hspi1, tx_data, rx_data, 3, 100);
+
+	    if(hal_status == HAL_OK)
+	    {
+		    //First byte of received data doesn't matter
+		    //First 5 bits (6th one is null) of second byte don't matter
+		    //All of third byte matters
+
+		    //Apply a mask (00000011) to just get last 2 bits of second byte
+		    secondByte = mask & rx_data[1];
+            thirdByte = rx_data[2];
+
+		    counts = (secondByte << 8) | thirdByte;
+		    //I am a little bit confused on how to convert the
+		    //digital value of the ADC to on-ounts
+		    //I decided to scale linearly from 3201 to 6270
+		    //which should keep the duty cycle between 5% and 10%
+		    counts = (counts * 3) + 3201;
```

Comment:
So, this MAY be right, and the reason I say may is because I don't know what you set your period to. BUT based on your calculations, if you get a value of 0 from the adc it corresponds to 5% of the period and according to your equation you would get a `counts` of 3201. So if that is 5% then your period is 3201 x 20 = 64020. If 64020 is in fact the period that you selected, then this equation is correct. Now, 3 and 3201 are "magic numbers" meaning I have no idea where you got them from and what they mean. It would be best to capture these values in a variable and use that instead.